app/server/contactUsApi.ts

```diff
@@ -0,0 +1,135 @@
+import { captureMessage } from "@sentry/node";
+import { Request, Response } from "express";
+import fetch from "node-fetch";
+import { contactUsConfig } from "../shared/contactUsConfig";
+import { ContactUsReq } from "../shared/contactUsTypes";
+import { getContactUsAPIHostAndKey } from "./apiGatewayDiscovery";
+import { log } from "./log";
+
+export const contactUsFormHandler = async (req: Request, res: Response) => {
+  const validBody = parseAndValidate(req.body);
+  if (!validBody) {
+    return res.status(400).send();
+  }
+
+  const apiConfig = await getContactUsAPIHostAndKey();
+  if (!apiConfig) {
+    const errorMessage = "Could not obtain contact-us-api host/key.";
+    log.error(errorMessage);
+    captureMessage(errorMessage);
+    return res.status(500).send();
+  }
+
+  fetch(apiConfig.host, {
+    method: "POST",
+    body: JSON.stringify(validBody),
+    headers: {
+      "Content-Type": "application/json",
+      "x-api-key": apiConfig.apiKey
+    }
+  })
+    .then(contactUsAPIResponse => {
+      if (!contactUsAPIResponse.ok) {
+        const errorMessage = `Unexpected error from contact-us-api endpoint. ${contactUsAPIResponse.status} ${contactUsAPIResponse.statusText}`;
+        log.error(errorMessage);
+        captureMessage(errorMessage);
+      }
+      res.status(contactUsAPIResponse.status).send();
+    })
+    .catch(error => {
+      const errorMessage =
+        "Unexpected error when trying to contact contact-us-api endpoint.";
+      log.error(errorMessage, error);
+      captureMessage(errorMessage);
+      res.status(500).send();
+    });
+};
+
+const parseAndValidate = (body: any): ContactUsReq | undefined => {
+  try {
+    const bodyAsJson = body ? JSON.parse(body) : "{}";
+
+    return validateContactUsFormBody(bodyAsJson)
+      ? buildContactUsReqBody(bodyAsJson)
+      : undefined;
+  } catch (error) {
+    return undefined;
+  }
+};
+
+const validateContactUsFormBody = (body: any): boolean =>
+  body &&
+  body.topic &&
+  validateTopics(body.topic, body.subtopic, body.subsubtopic) &&
+  body.name &&
+  body.email &&
+  isEmail(body.email) &&
+  body.subject &&
+  body.message;
+
+const validateTopics = (
+  reqTopic: string | undefined,
```

Comment:
Isn't that the same type as a `reqTopic?: string`

Reply:
Yes, it is. However the question mark implies that the argument is optional when it's not quite the case here. It's more that we don't know at this point what the contents of these variables are (or their types). But I suppose a better representation of that is to make this of type `unknown`. I will make this change now.